balancer/weightedtarget/weightedtarget.go

```diff
@@ -84,6 +84,17 @@ type weightedTargetBalancer struct {
 	targets map[string]Target
 }
 
+type localityKeyType string
+
+const localityKey = localityKeyType("locality")
+
+// GetLocality returns the locality from the resolver.State provided, or an
+// empty string if not present.
+func GetLocality(state resolver.State) string {
```

Comment:
Does this belong inside this LB policy or elsewhere?  Is it strictly a part of `weightedtarget`'s behavior or might other policies want to be able to set it?

Also let's avoid `Get`.

Reply:
I think it has to be part of this LB policy, as per this line in the gRFC: "To support the locality label in the WRR metrics, we will extend the weighted_target LB policy (see [A28](https://github.com/grpc/proposal/blob/master/A28-xds-traffic-splitting-and-routing.md)) to define a resolver attribute that indicates the name of its child."

Reply:
Renamed LocalityFromResolverState to keep it inline with what we called our helper to get XDSClient from resolver state.